webknossos/tests/test_dataset.py

```diff
@@ -34,14 +34,42 @@
     rmtree,
     snake_to_camel_case,
 )
+import subprocess
+import shlex
 
 from .constants import TESTDATA_DIR, TESTOUTPUT_DIR
 
+MINIO_ROOT_USER="TtnuieannGt2rGuie2t8Tt7urarg5nauedRndrur"
+MINIO_ROOT_PASSWORD="ANTN35UAENTS5UIAEATD"
+MINIO_PORT="8000"
+
+
+@pytest.fixture(autouse=True, scope="module")
+def docker_minio():
+    """Minio is an S3 clone and is used as local test server"""
+    container_name = "minio"
+    cmd = (
+        "docker run"
+        f" -p {MINIO_PORT}:9000"
+        f" -e MINIO_ROOT_USER={MINIO_ROOT_USER}"
+        f" -e MINIO_ROOT_PASSWORD={MINIO_ROOT_PASSWORD}"
+        f" --name {container_name}"
+        " --rm"
+        " -d"
+        " minio/minio server /data"
+    )
+    print("BEFORE", flush=True)
+    subprocess.check_output(shlex.split(cmd))
+    REMOTE_TESTOUTPUT_DIR.fs.mkdirs("testoutput", exist_ok=True)
+    yield
+    subprocess.check_output(["docker", "stop", container_name])
```

Comment:
Does this need try/finally or will it always call the docker stop?

Reply:
Good point, I think try-finally is needed.